### ISSUE TITLE
Actively detect links under the cursor when lines change

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#4323]
The PR https://github.com/xtermjs/xterm.js/pull/4298 made it so that an active link under the cursor would be re-evaluated instead of thrown away when the line changes, but  we also want to capture this case:

![Recording 2022-12-09 at 09 19 23](https://user-images.githubusercontent.com/2193314/206757526-85fd1b9b-3159-47af-a620-ac8f15d44ed7.gif)

Repro:

1. Fill viewport
2. Run `echo -n "http://example.com/" && sleep 1 && echo -n "a" && sleep 1 && echo -n "b" && sleep 1 && echo -n "c" && sleep 1 && echo -n " x" && sleep 1 && echo -n "x" && sleep 1 && echo "x" && sleep 1 && echo "y" && sleep 1 && echo "y" && sleep 1 && echo "y"`
3. Put cursor above `http://example.com/`, when a line is added such that the cursor is over the link, the link should get an underline and clicking should activate it, _without_  needing to move the cursor beforehand

This change makes this happen but will result in a lot of work:

```ts
    this._renderService.onRenderedViewportChange(e => {
      if (this._lastMouseEvent) {
        this._handleMouseMove(this._lastMouseEvent);
      }
    });
```

Can augment the above, the solution in #4298, some debouncing and possible some other form of caching to save some CPU cycles?

**Comment by jerch** [comment]
@Tyriar Yes the other PR partially fixes the old issue, but not yet for that particular edge case.

> Can augment the above, the solution in https://github.com/xtermjs/xterm.js/pull/4298, some debouncing and possible some other form of caching to save some CPU cycles?

I think so, but the devil might be in details here, how to get proper debouncing working from several terminal actions/states. (Have not yet checked how bad the perf would degrade w'o any debouncing...)

**Comment by Tyriar** [comment]
With debouncing we could make it quite small like 50ms as we're just trying to avoid it checking when constant parsing is happening, we would also only want to debounce these refreshes to keep regular mouse moves as responsive as possible as the delay would affect the underline showing up.
